src/hip_memory.cpp

```diff
@@ -2219,16 +2219,40 @@ hipError_t hipMemset(void* dst, int value, size_t sizeBytes) {
     return ihipLogStatus(ihipMemsetSync(dst, value, sizeBytes, nullptr, ihipMemsetDataTypeChar));
 }
 
+hipError_t ihipMemsetND(void* dst, size_t pitch, int value, size_t width, size_t height, size_t setHeight,size_t depth,
+                                              hipStream_t stream, enum ihipMemsetDataType copyDataType, bool async) {
+    size_t sizeBytes =0;
+    hipError_t hipStatus = hipSuccess;
+    if ((pitch == width) && (height == setHeight)) {
+        sizeBytes = pitch * setHeight * depth;
+        if(async)
+            return ihipMemsetAsync(dst, value, sizeBytes, stream, copyDataType);
+        else
+            return ihipMemsetSync(dst, value, sizeBytes, nullptr, copyDataType);
+    } else {
+        for(size_t i = 0; i < depth; ++i) {
+            for(size_t j = 0; j < setHeight; ++j) {
+                int* dstPtr = (int*) ((unsigned char*) dst + i * height * pitch + j * pitch);
```

Comment:
Is there any reason to cast this to an int*? Shouldn't it be void*?

Reply:
I have corrected this.